src/plugins/data/public/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -372,8 +374,8 @@ export class IndexPattern implements IIndexPattern {
           duplicateId,
           this.getConfig,
           this.savedObjectsClient,
-          this.patternCache,
-          this.fieldsFetcher
+          this.apiClient,
+          this.patternCache
```

Comment:
Can we improve the types for `patternCache` (both the class property and constructor argument uses `any`) to prevent this from happening in the future?